rsocket-core/src/main/java/io/rsocket/core/RSocketServer.java

```diff
@@ -262,10 +282,41 @@ public RSocketServer payloadDecoder(PayloadDecoder decoder) {
   public <T extends Closeable> Mono<T> bind(ServerTransport<T> transport) {
     return Mono.defer(
         new Supplier<Mono<T>>() {
-          ServerSetup serverSetup = serverSetup();
+          final ServerSetup serverSetup = serverSetup();
 
           @Override
           public Mono<T> get() {
+            int maxFrameLength = transport.maxFrameLength();
+
+            if (maxFrameLength > FRAME_LENGTH_MASK) {
+              return Mono.error(
+                  new IllegalArgumentException(
+                      "Configured maxFrameLength["
+                          + maxFrameLength
+                          + "] exits maxFrameLength limit "
+                          + FRAME_LENGTH_MASK));
+            }
+
+            if (maxFrameLength > maxPayloadSize) {
+              return Mono.error(
+                  new IllegalArgumentException(
+                      "Configured maxFrameLength["
+                          + maxFrameLength
+                          + "] exits maxPayloadSize["
+                          + maxPayloadSize
+                          + "]"));
+            }
+
+            if (mtu != 0 && mtu > maxFrameLength) {
+              return Mono.error(
+                  new IllegalArgumentException(
+                      "Configured maximumTransmissionUnit["
+                          + mtu
+                          + "] exits configured maxFrameLength["
+                          + maxFrameLength
+                          + "]"));
+            }
+
```

Comment:
Could this be encapsulated some place and shared with RSocketConnector? PayloadValidationUtils maybe?

Reply:
it is not directly related to Payload. But I agree that it is duplicated